tools/c7n_gcp/c7n_gcp/mu.py

```diff
@@ -46,7 +46,7 @@ def custodian_archive(packages=None, deps=()):
     requirements.update(deps)
     archive.add_contents(
         'requirements.txt',
-        generate_requirements(requirements, include_self=True))
+        generate_requirements(requirements, ignore=('setuptools'), include_self=True))
```

Comment:
this line looks suspect, ignore is a tuple, and its being set to a string, which means it likely wasn't doing anything.

Reply:
hmm yeah should be a tuple, but it was working. will change it though.

generate_requirements(requirements, ignore=('setuptools'), include_self=True))
```'cachetools==4.2.0\ncertifi==2020.12.5\nchardet==4.0.0\ngoogle-api-core==1.26.1\ngoogle-api-python-client==1.12.8\ngoogle-auth==1.24.0\ngoogle-auth-httplib2==0.1.0\ngoogleapis-common-protos==1.53.0\nhttplib2==0.19.0\nidna==2.10\njmespath==0.10.0\npackaging==20.9\nprotobuf==3.15.6\npyasn1==0.4.8\npyasn1-modules==0.2.8\npyparsing==2.4.7\npython-dateutil==2.8.1\npytz==2021.1\nratelimiter==1.2.0.post0\nrequests==2.25.1\nretrying==1.3.3\nrsa==4.6\nsix==1.15.0\nuritemplate==3.0.1\nurllib3==1.26.4'```

generate_requirements(requirements,  include_self=True))

```'cachetools==4.2.0\ncertifi==2020.12.5\nchardet==4.0.0\ngoogle-api-core==1.26.1\ngoogle-api-python-client==1.12.8\ngoogle-auth==1.24.0\ngoogle-auth-httplib2==0.1.0\ngoogleapis-common-protos==1.53.0\nhttplib2==0.19.0\nidna==2.10\njmespath==0.10.0\npackaging==20.9\nprotobuf==3.15.6\npyasn1==0.4.8\npyasn1-modules==0.2.8\npyparsing==2.4.7\npython-dateutil==2.8.1\npytz==2021.1\nratelimiter==1.2.0.post0\nrequests==2.25.1\nretrying==1.3.3\nrsa==4.6\nsetuptools==52.0.0.post20210125\nsix==1.15.0\nuritemplate==3.0.1\nurllib3==1.26.4'```

the problem is `setuptools==52.0.0.post20210125` was taken down so doesn't exist. it is not required for running custodian so i added to this ignore.